### PR TITLE
FTDCS-151 - Use aws-sdk official package to sign requests to AWS

### PR DESCRIPTION
**Pull request by JSRedondo**

### CODE REVIEW

File: package.json
Comment (rowanmanning) [inline package.json:1]: **suggestion:** can we revert the whitespace changes in this file, as it's noisy and will make the history hard to read.

File: lib/helpers/signed-aws-es-fetch.js
Comment (rowanmanning) [inline lib/helpers/signed-aws-es-fetch.js:26]: **question:** I was quite confused about how the whole file has changed, but I think maybe it's just that you have reorganised the file? It makes it quite difficult to understand what's actually changed without reorganising the file myself locally and diffing. How annoying would it be to change the order of the file back and commit it separately?

Comment (JSRedondo) [reply lib/helpers/signed-aws-es-fetch.js:26]: I just sorted it out to follow a logical order and to be easier to understand. Apparently, it wasn't a good idea.

Comment (JSRedondo) [reply lib/helpers/signed-aws-es-fetch.js]: It's basically the same code for some URL tools, I split the code to make it easier to check.

Comment (rowanmanning) [reply lib/helpers/signed-aws-es-fetch.js]: Heya, sorry I think there was a miscommunication here. I guess what I meant was that the purpose of this PR is to upgrade a dependency, however the PR also includes a lot of refactoring which is unrelated to the dependency bump. For example, the methods were reordered and reorganised.

What I really meant is that in order to review things quickly and easily, it's best if a PR contains _only_ the code related to the task at hand. In this case I want to be able to see just the changes which relate to the dependency upgrade rather than the refactoring, because when the code is rewritten and split across some new files it's much harder to go through line by line and check that the logic is still what we expect.

I'm not against refactoring, it's great! However it's better to have it as a separate PR or at least a separate commit so that it's clear what the intention of each unit of work is. Does that make sense? I struggled to know how to type this in a clear way 😅  so if a quick chat on a call would be easier then I'm happy to do that!

Comment (JSRedondo) [reply lib/helpers/signed-aws-es-fetch.js:81]: Well, there are some different functions than the original one because the implementation is not exactly the same.
Anyway, I reordered and renamed them with the original names.
Another PR can be done with proper names and the refactor.

File: package.json
Comment (rowanmanning) [reply package.json:1]: Oh weirdly this is still showing as the whole file being edited for me. It looks like we're still using tabs at the top level?

File: lib/helpers/signed-aws-es-fetch.js
Comment (rowanmanning) [inline lib/helpers/signed-aws-es-fetch.js:120]: **suggestion:** I might be reading this wrong, but I don't think this needs to be a promise-based function because `creds` is always just plan data. The [Constructor for `SignatureV4`](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/classes/_aws_sdk_signature_v4.signaturev4-1.html#constructor) accepts either a `Credentials` object _or_ a `Provider<Credentials>`, and we already have a credentials object so I think this is equivalent:

```suggestion
		credentials: creds,
```

File: lib/helpers/signed-aws-es-fetch.js
Comment (rowanmanning) [inline lib/helpers/signed-aws-es-fetch.js:107]: **nitpick:** could this be a default param?

```suggestion
async function signedFetch(url, opts, creds = {}) {
```

**Review by rowanmanning — CHANGES_REQUESTED** [review]
I think one or two smaller things but otherwise this is looking good 👍 do we want to pursue a review from Storytelling as code owners as well? Also unsure if Alex is part-way through a review

File: lib/helpers/signed-aws-es-fetch.js
Comment (JSRedondo) [reply lib/helpers/signed-aws-es-fetch.js:120]: I tested passing just the credentials and it didn't work. 
I tried first using plain credentials and later I changed for the async function. I don't know why but it's the way it is.

Comment (rowanmanning) [reply lib/helpers/signed-aws-es-fetch.js:120]: 🤷‍♂️  OK then, let's leave it

**Pull request merged**
